.github/workflows/hera_pr.yaml

```diff
@@ -41,6 +41,93 @@ jobs:
 
       - name: Run tests
         env:
+          ENABLE_BDIST_EXT_MODULE: 'yes'
           TOXENV: python${{ matrix.python-version }}
         run: |
           tox
+
+      - name: Check build
+        run: |
+          twine check .tox/dist/*
+
+      - name: upload build artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: pypi-build-arts
+          path: .tox/dist
```

Comment:
This seems to publish the constructed distributions as artifacts. To be clear, these are different than the actually published package, correct?

Reply:
Not quite. If we consolidate to a single GH workflow like I suggested, the `build` ( and test ) job will run as a dependency of `publish` job on push to `main` as well ( see more in https://github.com/argoproj-labs/hera-workflows/pull/101#issuecomment-1065169028 ), and thus provide us with the necessary Distribution builds ( wheels ) - As you can see _Distribution_ builds added to `setup.py` are `OS` + `Py` version specific.

HOWEVER ! what's usually done, is building `also` a `py-none-any` wheel + `tar.gz` and publishing it as well ( like `hera` publishes now ), e.g. ( from https://pypi.org/project/pydantic/ )

<img width="895" alt="Screen Shot 2022-03-11 at 4 44 18 PM" src="https://user-images.githubusercontent.com/63955749/157889557-e45dc269-51e4-4872-93cd-572ff5eb02be.png">